mr/src/main/java/org/opensearch/hadoop/rest/RestClient.java

```diff
@@ -520,7 +520,7 @@ public boolean indexExists(String index) {
     }
 
     private boolean exists(String indexOrType) {
-        Request req = new SimpleRequest(HEAD, null, indexOrType);
+        Request req = new SimpleRequest(GET, null, indexOrType);
```

Comment:
Hm ... the `HEAD` sounded right (we just need to check existence), why it was changed to `GET`?

Reply:
+1